### PR TITLE
KONFLUX-306: Test Konflux on OCP 4.14

### PR DESCRIPTION
**Pull request by gbenhaim**
Add an optional job for testing Konflux on OCP 4.14.

### CODE REVIEW

**Comment by gbenhaim** [comment]
/pj-rehearse pull-ci-redhat-appstudio-infra-deployments-main-appstudio-e2e-tests-ocp-4.14

**Comment by gbenhaim** [comment]
depends on https://github.com/openshift/release/pull/48417

**Comment by gbenhaim** [comment]
The rehearsal job fails with:

```
could not run steps: step appstudio-e2e-tests-ocp-4.14 failed: "appstudio-e2e-tests-ocp-4.14" pre steps failed: failed to create or restart appstudio-e2e-tests-ocp-4.14-ipi-install-rbac pod: unable to create pod: Pod "appstudio-e2e-tests-ocp-4.14-ipi-install-rbac" is invalid: [spec.volumes[11].name: Invalid value: "appstudio-e2e-tests-ocp-4.14": must not contain dots, spec.containers[0].volumeMounts[6].name: Not found: "appstudio-e2e-tests-ocp-4.14"] 
```

File: ci-operator/config/redhat-appstudio/infra-deployments/redhat-appstudio-infra-deployments-main.yaml [hunk 1]
```diff
@@ -53,6 +53,22 @@ tests:
     test:
     - ref: redhat-appstudio-e2e
     workflow: redhat-appstudio-claim
+- as: appstudio-e2e-tests-ocp-4.14
```
Comment (psturc) [inline ci-operator/config/redhat-appstudio/infra-deployments/redhat-appstudio-infra-deployments-main.yaml:56]:
```suggestion
- as: appstudio-e2e-tests-ocp-4-14
```

Comment (gbenhaim) [reply ci-operator/config/redhat-appstudio/infra-deployments/redhat-appstudio-infra-deployments-main.yaml]: fixed

**Comment by gbenhaim** [comment]
@psturc I think that we are good to go.

**Pull request merged**